src/wallet/rpcwallet.cpp

```diff
@@ -1769,12 +1838,17 @@ UniValue gettransaction(const UniValue& params, bool fHelp)
         if(params[1].get_bool())
             filter = filter | ISMINE_WATCH_ONLY;
 
+    std::optional<int> asOfHeight;
+    if (params.size() > 2) {
+        asOfHeight = params[2].get_int();
+    }
+
     UniValue entry(UniValue::VOBJ);
     if (!pwalletMain->mapWallet.count(hash))
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid or non-wallet transaction id");
     const CWalletTx& wtx = pwalletMain->mapWallet[hash];
 
-    CAmount nCredit = wtx.GetCredit(filter);
+    CAmount nCredit = wtx.GetCredit(std::nullopt, filter);
```

Comment:
Add a comment for why this passes nullopt.

Reply:
Should have been `asOfHeight`. Fixed.